eman2/protocols/protocol_tomo_template_match.py

```diff
@@ -0,0 +1,201 @@
+# *
+# * Authors:     David Herreros
+# *
+# * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC
+# *
+# * This program is free software; you can redistribute it and/or modify
+# * it under the terms of the GNU General Public License as published by
+# * the Free Software Foundation; either version 2 of the License, or
+# * (at your option) any later version.
+# *
+# * This program is distributed in the hope that it will be useful,
+# * but WITHOUT ANY WARRANTY; without even the implied warranty of
+# * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# * GNU General Public License for more details.
+# *
+# * You should have received a copy of the GNU General Public License
+# * along with this program; if not, write to the Free Software
+# * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+# * 02111-1307  USA
+# *
+# *  All comments concerning this program package may be sent to the
+# *  e-mail address 'scipion@cnb.csic.es'
+# *
+# **************************************************************************
+
+import os
+import glob
+import ntpath
+
+from tomo.objects import SetOfTomograms
+from pyworkflow import utils as pwutils
+from pyworkflow.protocol.params import (PointerParam, IntParam,
+                                        BooleanParam, LEVEL_ADVANCED,
+                                        StringParam, FloatParam)
+from pyworkflow.em.convert import ImageHandler
+from pyworkflow.em.data import Volume
+from pyworkflow.utils.path import moveFile, cleanPath, makePath
+
+import eman2
+from eman2.convert import loadJson
+
+from tomo.protocols import ProtTomoPicking
+from tomo.objects import Coordinate3D, SetOfCoordinates3D
+
+
+class EmanProtTempMatch(ProtTomoPicking):
+    """
+    This protocol wraps *e2spt_tempmatch.py* EMAN2 program.
+
+    It will perform a sweep of an initial volume against a tomogram
+    to find correlation peaks and extract the corresponding subtomogram
+    coordinates
+
+    """
+
+    _label = 'template matching'
+
+    def __init__(self, **args):
+        ProtTomoPicking.__init__(self, **args)
+
+    # --------------------------- DEFINE param functions ----------------------
+
+    def _defineParams(self, form):
+        form.addSection(label='Input')
+        form.addParam('inputSet', PointerParam,
+                      pointerClass='SetOfTomograms',
+                      label="Input tomograms", important=True,
+                      help='Specify tomograms containing reference-like particles\n'
+                           'to be extracted')
+        form.addParam('ref', PointerParam, pointerClass="Volume",
+                      label='Reference volume',
+                      help='Specify a 3D volume')
+        form.addParam('nptcl', IntParam, default=500,
+                      label='Number of particles',
+                      help='Maximum number of particles')
+        form.addParam('dthr', FloatParam, default=-1,
```

Comment:
Makes it sense to provide some sort of help here?

Reply:
All the help messages for the input parameters are based on the information provided in the program of Template Marching, but some of them do not have any information about their usage